### DISCUSSION TITLE
compile error when I do 'cargo build'

### DISCUSSION THREAD

**Discussion by Lempossible** [vectordotdev/vector#17640] [Q&A] [answered]
### A note for the community

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Problem

when I execute ‘cargo build’, it reports an error.
"[failed to run custom build command for `vector-core v0.1.0](error: failed to run custom build command for `vector-core v0.1.0)"

Caused by:
  process didn't exit successfully: `/vector_back/target/debug/build/vector-core-ee2ba90d479c8fb4/build-script-build` (exit status: 101)
  --- stdout
  cargo:rerun-if-changed=proto/event.proto
 --- stderr
  thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Custom { kind: Other, error: "protoc failed: Unknown flag: --experimental_allow_proto3_optional\n" }', lib/vector-core/build.rs:8:10
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

### Version

master branch

**Comment by spencergilbert** [comment, marked as answer]
Do you have `protoc` installed? You can see [this script](https://github.com/vectordotdev/vector/blob/master/scripts/environment/install-protoc.sh) that we use to install `protoc` in CI.

**Comment by sinkingpoint** [comment]
I found this compiling on old debian releases (e.g. buster) - the version of protoc in their repos doesn't support the `experimental_allow_proto3_optional` flag. Using a bullseye/bookworm base image worked

**Comment by jmagnuson** [comment]
If making system-wide changes isn't feasible or desired, `prost-build` alternatively [accepts](https://github.com/tokio-rs/prost/tree/master/prost-build#protoc) a custom path to `protoc` via the `PROTOC` environment variable. This is what I did when working on a shared system:

```shell
scripts/environment/install-protoc.sh "${HOME}/bin"

PROTOC="${HOME}/bin/protoc" cargo build ...
```
